### PR TITLE
modem-manager: Add signed-payload metadata for some Quectel devices

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/modem-manager/fu-mm-device.c [hunk 1]
```diff
@@ -196,6 +196,41 @@ validate_firmware_update_method(MMModemFirmwareUpdateMethod methods, GError **er
...
 
+static void
+fu_mm_device_probe_quectel_flags(FuMmDevice *self)
+{
+	const gchar *version = fu_device_get_version(FU_DEVICE(self));
+	g_autofree gchar *name = NULL;
+	struct {
+		const gchar *name;
+		const gchar *version;
+	} secboot[] = {{"EM05GF", "EM05GFAR07A07M1G_01.005.01.005"},
```
Comment (aleksander0m) [inline plugins/modem-manager/fu-mm-device.c:207]: I'm sure you're not happy with having a hardcoded list; I guess there's no better way to handle this, right?

Comment (hughsie) [reply plugins/modem-manager/fu-mm-device.c:207]: Ya I'm not 100% happy, but apparently there's no way of asking the firmware "_hey are you signed_" -- at least not yet. Even some heuristic on the version string would be good, but alas.

Comment (superm1) [reply plugins/modem-manager/fu-mm-device.c:207]: Are those earlier versions preproduction presumably? Maybe it's TOO much detail?
Especially if downgrades aren't supported.

Comment (superm1) [reply plugins/modem-manager/fu-mm-device.c:207]: This has me questioning the efficacy of such an effort. If the boot rom isn't signed and the first payload you had in there wasn't signed how can you guarantee the safety of a payload that introduced signing?

Maybe that's too deep a question in root of trust and we should just be happy it's signed now and move on.

Comment (hughsie) [reply plugins/modem-manager/fu-mm-device.c:207]: The new firmware blows an eFuse!

**Pull request merged**